js/base-engine.js

```diff
@@ -1254,6 +1268,62 @@ function d20plusEngine () {
 		})
 	};
 
+	d20plus.engine.objectsStashProps = (obj, state) => {
+		const props = [
+			"emits_bright_light",
+			"emits_low_light",
+			"has_directional_bright_light",
+			"has_directional_dim_light",
+			"showplayers_bar1",
+			"showplayers_bar2",
+			"showplayers_bar3",
+			"showname",
+		];
+		props.each((prop) => {
+			if (!state) {
+				if (obj.attributes[prop]) {
+					obj.attributes[`bR20_${prop}`] = true;
+					obj.attributes[prop] = false;
+				}
+			} else {
+				if (obj.attributes[`bR20_${prop}`]) {
+					obj.attributes[prop] = true;
+					obj.attributes[`bR20_${prop}`] = false;
+					delete obj.attributes[`bR20_${prop}`];
+				}
+			}
+		});
+	}
+
+	d20plus.engine.objectsHideUnhide = (query, val, prefix, state) => {
+		let some = false;
+		for (const o of d20.engine.canvas._objects) {
+			const model = o.model;
+			if (!model) continue;
+			if (`${model.get(query)}`.search(val) > -1) {
+				const l = model.attributes.layer;
```

Comment:
prefer e.g. `const {layer} = model.attributes` to randomly one-letter-renaming vars